From 9d914631757f42055d2bacbab9c4bfad0765a3d0 Mon Sep 17 00:00:00 2001
From: root <root@zeus.(none)>
Date: Mon, 18 Mar 2013 09:16:38 +0200
Subject: [PATCH 1/6] Added ALC899

---
 kernel/drv/oss_hdaudio/hdaudio_codecids.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/drv/oss_hdaudio/hdaudio_codecids.h b/kernel/drv/oss_hdaudio/hdaudio_codecids.h
index 67670ab..f50b0d2 100644
--- a/kernel/drv/oss_hdaudio/hdaudio_codecids.h
+++ b/kernel/drv/oss_hdaudio/hdaudio_codecids.h
@@ -836,6 +836,7 @@ static const codec_desc_t codecs[] = {
   {0x10ec0888, "ALC888", VF_ALC88X_HACK, (char **) &alc883remap}, 
   {0x10ec0889, "ALC889", VF_ALC88X_HACK, (char **) &alc883remap}, 
   {0x10ec0892, "ALC892", VF_ALC88X_HACK, (char **) &alc883remap}, 
+  {0x10ec0899, "ALC899", VF_ALC88X_HACK, (char **) &alc883remap},
 
   /* CMedia HDA codecs */
   {0x13f69880, "CMI9880", VF_NONE, (char **) &cmi9880remap},
-- 
1.8.3.2

